avocado/core/job.py

```diff
@@ -515,6 +527,9 @@ def run(self):
             if not settings.get_value('runner.behavior', 'keep_tmp_files',
                                       key_type=bool, default=False):
                 data_dir.clean_tmp_files()
+            if self.time_end == -1:
+                self.time_end = time.time()
+                self.time_elapsed = self.time_end - self.time_start
```

Comment:
Nitpick: `time_elapsed` is overwritten when `time_end` is not set. But users could set the `time_elapsed` without setting `time_end`, or they might want to set `time_end` and assume the `time_elapsed` would be calculated automatically.

Anyway do you really see some users to need to ever overwrite this? I'd be fine with making it automatic and if we ever get a request, we could always make it conditional.